### PR TITLE
Fix actionsgroups API example response and cleanup topic generally

### LINKED ISSUE #3751: [DOC] Update security plugin API `actiongroups` sample response
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

Update the documentation for the security plugin default action-groups API example:
[Current example response](https://t.corp.amazon.com/V861844626): 
```json
{
  "read": {
    "reserved": true,
    "hidden": false,
    "allowed_actions": [
      "indices:data/read*",
      "indices:admin/mappings/fields/get*"
    ],
    "type": "index",
    "description": "Allow all read operations",
    "static": true
  },
  ...
}
```
<a name="response"></a>
Updated example response:
```json
{
  "read": {
    "reserved": true,
    "hidden": false,
    "allowed_actions": [
      "indices:data/read*",
      "indices:admin/mappings/fields/get*",
      "indices:admin/resolve/index"
    ],
    "type": "index",
    "description": "Allow all read operations",
    "static": true
  },
  ...
}
```
 
This change updates the allowed actions to reflect the correct response being sent.

To test this:
1. Download [latest](https://opensearch.org/samples/docker-compose.yml) docker-compose.yml
2. Run `docker-compose up` from the download location
3. Goto OpenSearch-Dashboards window in the browser and navigate to Dev Tools
4. Run `GET _plugins/_security/api/actiongroups/read`
5. You should see the updated response there

### PR DESCRIPTION
**Pull request by cwillum**
### Description
The actiongroups API GET call that retrieves all actions groups includes an example response in need of an update. This whole topic could benefit from the addition of copy/copy curl tags for requests, capitalization of "Security plugin" instances, and language tags for the basic JSON code examples for requests.  

### Issues Resolved
Made changes based on description above.
Fixes #3751 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _security/access-control/api.md [hunk 1]
```diff
@@ -1529,8 +1592,9 @@ Using the PATCH method also requires a user to have a security configuration tha
 ```bash
 curl -X PATCH -k -i --cert <admin_cert file name> --key <admin_cert_key file name> <domain>/_opendistro/_security/api/audit -H 'Content-Type: application/json' -d'[{"op":"add","path":"/config/enabled","value":"true"}]'
 ```
+{% include copy-curl.html %}
 
-The OpenSearch Dashboards dev tool does not currently support the PATCH method. You can use [curl](https://curl.se/), [Postman](https://www.postman.com/), or another alternative process to update the configuration using this method. To follow the GitHub issue for support of the PATCH method in Dashboards, see [issue #2343](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2343).
+The OpenSearch Dashboards Dev tool does not currently support the PATCH method. You can use [curl](https://curl.se/), [Postman](https://www.postman.com/), or another alternative process to update the configuration using this method. To follow the GitHub issue for support of the PATCH method in Dashboards, see [issue #2343](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2343).
```
Comment (kolchfa-aws) [inline _security/access-control/api.md:1597]:
```suggestion
OpenSearch Dashboards Dev Tools do not currently support the PATCH method. You can use [curl](https://curl.se/), [Postman](https://www.postman.com/), or another alternative process to update the configuration using this method. To follow the GitHub issue for support of the PATCH method in Dashboards, see [issue #2343](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2343).
```

**Review by kolchfa-aws — APPROVED** [review]
LGTM. Thanks for cleaning this up, Chris! You can also add {% include copy.html %} after yaml file settings.

**Comment by cwillum** [comment]
@DarshitChanpura This fixes the "get all action groups" API example response you identified in #3751. If you want to have a look, you can find the changes starting on line 218. I also added a second action group in the example to show you get more than one with this GET call. But the "read" action group includes the additional line you pointed out. The other changes are things I found across the whole topic -- a little cleanup.
Thanks for creating the detailed issue. Made this easy.

**Review by hdhalter — COMMENTED** [review]
Chris, can you please add a redirect under nav_order? Thanks.

```
redirect_from: 
 - /security-plugin/access-control/api/
```

**Review by hdhalter — APPROVED** [review]

File: _security/access-control/api.md [hunk 2]
```diff
@@ -169,12 +179,17 @@ Introduced 1.0
 
 Retrieves one action group.
 
+```json
+GET _plugins/_security/api/actiongroups/<action-group>
+```
+{% include copy-curl.html %}
 
-#### Request
+#### Example request
```
Comment (DarshitChanpura) [inline _security/access-control/api.md:187]: not sure if this was intended. I see the format to be `Request` and `Example Response` in other occurences. @cwillum Thoughts on reverting it to `Request`?

File: _security/access-control/api.md [hunk 3]
```diff
@@ -218,12 +233,23 @@ GET _plugins/_security/api/actiongroups/
...
       "indices:data/read*",
-      "indices:admin/mappings/fields/get*"
+      "indices:admin/mappings/fields/get*",
+      "indices:admin/resolve/index"
     ],
     "type": "index",
     "description": "Allow all read operations",
     "static": true
   },
+  "cluster_all": {
```
Comment (DarshitChanpura) [inline _security/access-control/api.md:243]: nicely done.. This should help users understand that there are more than 1 actiongroups in the cluster.

File: _security/access-control/api.md [hunk 4]
```diff
@@ -235,9 +261,10 @@ Introduced 1.0
 
 #### Request
 
-```
+```json
```
Comment (DarshitChanpura) [inline _security/access-control/api.md:264]: qq, does changing the format to json help with the layout? I was fiddling around with this when I created the issue with `json`, `shell`, `code`, etc. but it just ended up being displayed as a normal code block.

**Review by DarshitChanpura — COMMENTED** [review]
Thank you for fixing the doc @cwillum !

File: _security/access-control/api.md (hunk 4 above)
Comment (kolchfa-aws) [reply _security/access-control/api.md:264]: @DarshitChanpura It doesn't help with the layout, but it does provide syntax highlighting.

Comment (cwillum) [reply _security/access-control/api.md:264]: Yeah, good question. The team just started using a linter (Vale) to help analyze style issues in the file. Eventually the linter/code analyzer will flag a code block if it doesn't have a language associated with it. So I added the `json` language to the examples to get ahead of what's coming down the road.

File: _security/access-control/api.md (hunk 2 above)
Comment (cwillum) [reply _security/access-control/api.md:187]: Good catch, DC. Technically, this is an example request: it contains a real name for an action group. And the Example response contains the same action group name. All of these should have a proper example for the request and the response. In that case, they would be example requests and example responses. But the format here uses a generic filler <action-group> for most of these requests, which isn't a proper example. For consistency I'll change this back to Request. Thanks.

**Pull request merged**